src/injector.ts

```diff
@@ -63,6 +72,12 @@ export class Injector {
     provide(token : any, defaultValue? : any): any {
         let hasDefault = arguments.length > 1;
 
+        // TODO: ask @rezonant if this is safe
+        if( token instanceof ReflectedTypeRef && token.isGeneric()){
```

Comment:
Checking if instanceof `ReflectedTypeRef` is fine, and I think its definitely what we'll need to do. According to my above comment though, we should be able to remove the `isGeneric()` check to allow for all type references, providing some interesting implications.